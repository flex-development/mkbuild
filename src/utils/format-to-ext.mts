/**
 * @file Utilities - formatToExt
 * @module mkbuild/utils/formatToExt
 */

import type { Format } from '@flex-development/mkbuild'
import type { Ext } from '@flex-development/pathe'

/**
 * Get a file extension for a module `format`.
 *
 * @see {@linkcode Ext}
 * @see {@linkcode Format}
 *
 * @category
 *  utils
 *
 * @this {void}
 *
 * @param {Format} format
 *  Module format
 * @return {Ext}
 *  File extension for `format`
 */
function formatToExt(this: void, format: Format): Ext {
  // note: if more module formats are added, the switch statement below will
  // result in a lint error (@typescript-eslint/switch-exhaustiveness-check)
  switch (format) {
    case 'cjs':
      return '.cjs'
    case 'esm':
      return '.mjs'
    case 'iife':
    case 'system':
    case 'umd':
      return '.js'
  }
}

export default formatToExt
