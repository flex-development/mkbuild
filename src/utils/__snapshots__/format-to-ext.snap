// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:utils/formatToExt > should return file extension for \`format\` ("cjs") 1`] = `".cjs"`;

exports[`unit:utils/formatToExt > should return file extension for \`format\` ("esm") 1`] = `".mjs"`;

exports[`unit:utils/formatToExt > should return file extension for \`format\` ("iife") 1`] = `".js"`;

exports[`unit:utils/formatToExt > should return file extension for \`format\` ("system") 1`] = `".js"`;

exports[`unit:utils/formatToExt > should return file extension for \`format\` ("umd") 1`] = `".js"`;
